2020-12-19 20:59:08,775 ERROR: Exception on /edit_address [GET] [in /Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/kustarddev/Desktop/microblog/app/routes.py", line 206, in edit_address
    return render_template('edit_address.html', title=_('Edit Address'),
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/templating.py", line 137, in render_template
    return _render(
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/kustarddev/Desktop/microblog/app/templates/edit_address.html", line 2, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
  File "/Users/kustarddev/Desktop/microblog/app/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block "doc"
    {%- block html %}
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block "html"
    {% block body -%}
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block "body"
    {% block content -%}
  File "/Users/kustarddev/Desktop/microblog/app/templates/base.html", line 48, in block "content"
    {% block app_content %}{% endblock %}
  File "/Users/kustarddev/Desktop/microblog/app/templates/edit_address.html", line 16, in block "app_content"
    {% include '_address.html' %}
  File "/Users/kustarddev/Desktop/microblog/app/templates/_address.html", line 18, in top-level template code
    <p><a href="{{ url_for('edit') }}">{{ _('Edit your address') }}</a></p>
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/helpers.py", line 370, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 2216, in handle_url_build_error
    reraise(exc_type, exc_value, tb)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/helpers.py", line 357, in url_for
    rv = url_adapter.build(
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/werkzeug/routing.py", line 2179, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'edit'. Did you forget to specify values ['address_id']?
2020-12-19 20:59:52,311 ERROR: Exception on /edit_address [GET] [in /Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/kustarddev/Desktop/microblog/app/routes.py", line 206, in edit_address
    return render_template('edit_address.html', title=_('Edit Address'),
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/templating.py", line 137, in render_template
    return _render(
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/kustarddev/Desktop/microblog/app/templates/edit_address.html", line 2, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
  File "/Users/kustarddev/Desktop/microblog/app/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block "doc"
    {%- block html %}
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block "html"
    {% block body -%}
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block "body"
    {% block content -%}
  File "/Users/kustarddev/Desktop/microblog/app/templates/base.html", line 48, in block "content"
    {% block app_content %}{% endblock %}
  File "/Users/kustarddev/Desktop/microblog/app/templates/edit_address.html", line 16, in block "app_content"
    {% include '_address.html' %}
  File "/Users/kustarddev/Desktop/microblog/app/templates/_address.html", line 18, in top-level template code
    <p><a href="{{ url_for('edit') }}">{{ _('Edit your address') }}</a></p>
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/helpers.py", line 370, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 2216, in handle_url_build_error
    reraise(exc_type, exc_value, tb)
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/flask/helpers.py", line 357, in url_for
    rv = url_adapter.build(
  File "/Users/kustarddev/Desktop/microblog/venv/lib/python3.9/site-packages/werkzeug/routing.py", line 2179, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'edit'. Did you forget to specify values ['address_id']?
